giraffe/src/utils/legend/staticLegend.ts

```diff
@@ -138,16 +138,17 @@ export const convertLineSpec = (
   }
 
   const fillColumns = columnKeys.map(key => {
-    const column: string[] = mappings.map(fillColumn => {
+    const fillColumn: string[] = sortOrder.map(index => {
+      const columns = mappings[`${fillIndices[index]}`]
       const fillFormatter = getColumnFormatter(key)
-      return fillFormatter(fillColumn[key])
+      return fillFormatter(columns[key])
     })
 
     return {
       key,
       name: key,
       type: spec.table.getColumnType(key),
-      values: column,
+      values: fillColumn,
       colors,
     }
   })
```

Comment:
is this function unit testable? seems like a good candidate for some unit tests.

Reply:
It is, but an assumption has to be made. We have to assume that line graphs will sort the legend by value.

The reason there hasn't been unit tests for this previously was because line graphs (in overlaid mode) were not sorted in the tooltip (or legend as we call it now). This was quietly changed with the initial implementation of Static Legend with multiple components now consuming the underlying legend component.

I think it's time to codify the assumption and as you suggested, add some unit tests for this.